api-common-java/src/main/java/com/google/api/core/ApiFutures.java

```diff
@@ -159,57 +232,89 @@ public static <V, X> ApiFuture<X> transform(
             executor));
   }
 
+  /**
+   * Creates a new {@code ApiFuture} whose value is a list containing the values of all its input
+   * futures, if all succeed.
+   *
+   * <p>The list of results is in the same order as the input list.
+   *
+   * <p>Note that this method is a delegate of {@link Futures#allAsList(Iterable)}.
+   *
+   * @param futures Futures to combine
+   * @return A future that provides a list of the results of the component futures
+   * @see Futures#allAsList(Iterable)
+   */
   public static <V> ApiFuture<List<V>> allAsList(
       Iterable<? extends ApiFuture<? extends V>> futures) {
     return new ListenableFutureToApiFuture<>(
         Futures.allAsList(
             Iterables.transform(
                 futures,
-                new Function<ApiFuture<? extends V>, ListenableFuture<? extends V>>() {
-                  public ListenableFuture<? extends V> apply(ApiFuture<? extends V> apiFuture) {
-                    return listenableFutureForApiFuture(apiFuture);
-                  }
-                })));
+                (Function<ApiFuture<? extends V>, ListenableFuture<? extends V>>)
+                    ApiFutures::listenableFutureForApiFuture)));
   }
 
+  /**
+   * Creates a new {@code ApiFuture} whose value is a list containing the values of all its
+   * successful input futures. The list of results is in the same order as the input list, and if
+   * any of the provided futures fails or is canceled, its corresponding position will contain
+   * {@code null} (which is indistinguishable from the future having a successful value of {@code
+   * null}).
+   *
+   * <p>The list of results is in the same order as the input list.
+   *
+   * <p>Note that this method is a delegate of {@link Futures#successfulAsList(Iterable)}.
+   *
+   * @param futures Futures to combine
+   * @return A future that provides a list of the results of the component futures
+   * @see Futures#successfulAsList(Iterable)
+   */
   @BetaApi
   public static <V> ApiFuture<List<V>> successfulAsList(
       Iterable<? extends ApiFuture<? extends V>> futures) {
     return new ListenableFutureToApiFuture<>(
         Futures.successfulAsList(
             Iterables.transform(
                 futures,
-                new Function<ApiFuture<? extends V>, ListenableFuture<? extends V>>() {
-                  public ListenableFuture<? extends V> apply(ApiFuture<? extends V> apiFuture) {
-                    return listenableFutureForApiFuture(apiFuture);
-                  }
-                })));
-  }
-
-  /*
-   * @deprecated Use {@linkplain #transformAsync(ApiFuture, ApiFunction, Executor) the
-   * overload that requires an executor}. For identical behavior, pass {@link
-   * com.google.common.util.concurrent.MoreExecutors#directExecutor}, but consider whether
-   * another executor would be safer.
+                (Function<ApiFuture<? extends V>, ListenableFuture<? extends V>>)
```

Comment:
Please remove the cast. It shouldn't be needed.

Reply:
This is outdated.